### PR TITLE
Implement memory.grow and memory.current

### PR DESCRIPTION
**Pull request by pepyakin**
Still uses `ExternalName::testcase` hack though.

### CODE REVIEW

File: lib/execute/src/memory.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        Self {
+            mmap,
+            current: initial,
+            maximum,
+        }
+    }
+
+    /// Returns an base address of this linear memory.
+    pub fn base_addr(&self) -> *mut u8 {
+        self.mmap.as_ptr() as *mut u8
```
[the remark this answers is not in the document]
Comment (pepyakin) [reply lib/execute/src/memory.rs]: yeah, sure. I thought that this is unnecessary limitation, but now I think that this doesn't really matter.

File: filetests/grow.wat [hunk 2]
```diff
@@ -0,0 +1,31 @@
+(module
+  (memory 1)
+  (func $assert (param i32)
+    (block $ok
+      (br_if $ok
+        (i32.ne
+          (i32.const 0)
+          (get_local 0)
+        )
```
[the remark this answers is not in the document]
Comment (pepyakin) [reply filetests/grow.wat]: you right, thanks!

File: lib/environ/src/environ.rs [hunk 3]
```diff
@@ -377,7 +377,15 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
...
                 argument_bytes: None,
-                params: vec![AbiParam::new(I32)],
+                params: vec![
+                    AbiParam::new(I32),
+                    AbiParam {
+                        value_type: self.pointer_type(),
+                        purpose: ArgumentPurpose::VMContext,
+                        extension: ArgumentExtension::None,
+                        location: ArgumentLoc::Unassigned,
+                    },
```
[the remark this answers is not in the document]
Comment (pepyakin) [reply lib/environ/src/environ.rs]: thanks will do!

[the remark this answers is not in the document]
Comment (pepyakin) [reply lib/environ/src/environ.rs]: [Done](https://github.com/CraneStation/wasmtime/pull/9/commits/7450fd30fd8aec629ad88eed1bda92efec24551b)!

File: lib/execute/src/memory.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pepyakin) [reply lib/execute/src/memory.rs]: [Ready](https://github.com/CraneStation/wasmtime/pull/9/commits/d84012f0b74b87485272a3798f7c36d9021f5080)!

File: filetests/grow.wat (hunk 2 above)
[the remark this answers is not in the document]
Comment (pepyakin) [reply filetests/grow.wat]: [Simplified](https://github.com/CraneStation/wasmtime/pull/9/commits/09f94e52cb3021cbc40a5bc071f91efe60014cb9)!

**Comment by sunfishcode** [comment]
This is a good step forward, we can continue to iterate on it in the tree.

**Pull request merged**